blockindex/sgd_indexer.go

```diff
@@ -285,9 +285,21 @@ func (sgd *sgdRegistry) StartHeight() uint64 {
 
 // PutBlock puts a block into SGDIndexer
 func (sgd *sgdRegistry) PutBlock(ctx context.Context, blk *block.Block) error {
-	if blk.Height() < sgd.startHeight {
+	tipHeight, err := sgd.Height()
```

Comment:
wrap into a subfunc

Reply:
wrap a function to calculate `expectHeight`